lib/compiler/decorators.ts

```diff
@@ -52,30 +50,46 @@ export function managed(
 export function meta(_options: {
   singleton?: boolean;
   scope?: string;
-}): ClassDecorator {
+  eager?: boolean;
+}): any {
   return () => undefined;
 }
 
+export function provides(target: object, propertyKey: string): void;
+/**
+ * @deprecated
+ */
+export function provides(_option?: { singleton?: boolean }): MethodDecorator;
 export function provides(_option?: { singleton?: boolean }): MethodDecorator {
   return () => undefined;
 }
 
-export function unit(_target: any): void {
-  //
+export function unit(target: any): void {
+  addTsdiMarker(target);
 }
 
-export function container(_config: { units: any[] }): any {
-  return () => undefined;
+export function container(config: { units: any[] }): ClassDecorator;
+// tslint:disable-next-line:ban-types
+export function container<T extends Function>(target: T): void;
+export function container(config: any): ClassDecorator | void {
+  if (typeof config !== 'function') {
+    return () => undefined;
+  }
 }
 
-export function initialize(target: object, propertyKey: string): void;
-export function initialize(): MethodDecorator;
-export function initialize(): MethodDecorator | void {
+export function postConstruct(target: object, propertyKey: string): void;
+export function postConstruct(): MethodDecorator;
```

Comment:
I will rename them then. I would as well favor of keeping `initialize` and `destroy` but since these decorators have runtime side-effects from version 0.x I need to have a new pair.
The compiler version is nearly free of a runtime and has very much less overhead during container creation.